File: app/src/main/java/org/freelesson/pagingsample/db/RepoDatabase.java
package org.freelesson.pagingsample.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import org.freelesson.pagingsample.model.Repo;

@Database(entities = {Repo.class}, version = 1, exportSchema = false)
public  abstract class RepoDatabase extends RoomDatabase  {
    public abstract RepoDao repoDaos();

    private static RepoDatabase INSTANCE;
    public static RepoDatabase getInstance(Context context) {
        if (INSTANCE==null){
            INSTANCE = buildDatabase(context);
        }
        return INSTANCE;
    }
    private static RepoDatabase buildDatabase(Context context) {
        return Room.databaseBuilder(context.getApplicationContext(),RepoDatabase.class,"Github.db").build();
    }
}

File: app/src/main/kotlin/org/freelesson/pagingsample/db/RepoDatabase.kt
package org.freelesson.pagingsample.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import org.freelesson.pagingsample.model.RemoteKeys
import org.freelesson.pagingsample.model.Repo

@Database(entities = [Repo::class, RemoteKeys::class], version = 1, exportSchema = false)
abstract class RepoDatabase : RoomDatabase() {
    abstract fun repoDao(): RepoDao
    abstract fun remoteKeysDao(): RemoteKeysDao

    companion object {
        @Volatile
        private var INSTANCE: RepoDatabase? = null
        fun getInstance(context: Context): RepoDatabase = INSTANCE ?: synchronized(this) {
            INSTANCE ?: buildDatabase(context).also {
                INSTANCE = it
            }
        }

        private fun buildDatabase(context: Context): RepoDatabase {
            return Room.databaseBuilder(
                context.applicationContext,
                RepoDatabase::class.java,
                "Github.db"
            ).build()
        }
    }
}